docs/proposals/ai-studio.md

```diff
@@ -0,0 +1,97 @@
+# Motivation
+
+Today, there is no notion of ordering between the containers. But we know that we have a dependency between
+the client application and the container that is running the model.
+
+The second issue is that there is no concept of starting point for a container so today we rely only on the
+container being started by the container engine and we know that this is not adequate for the model service container
+
+So this is handle by a kind of dirty fix: the containers are all started in parallel but as the client application
+will fail because the model service is started (as it take a while), so we are trying to restart the client application
+until the model service is properly started.
+
+The purpose of this change is to propose an update to the ai-studio.yaml so that it is as much generic as it
+could be and inspired from the Compose specification.
+
+## Proposed changes
+
+Define a dependency between containers: so in the definition of container, we would add a new ```depends``` field
+that would define the ordering between the containers. This would be an optional field.
+
+Define a condition for the container to be properly started: this would be based on the HEALTHCHECK that can already
+be defined in a Dockerfile. In the first iteration, we would support only the ```test``` field. If ```heathcheck``` is defined,
+then we would check for the healthcheck status field to be ```healthy```
+
```

Comment:
The ```test``` command is intended to be executed from inside the container in which it is defined.

Reply:
I also think we should simply use the official definition of Kubernetes for readiness and liveness, and use that in the pod for each contain.

See https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/

Reply:
Yes that would be an option but we would support only the exec variant as it would be based on podman healthcheck implementation